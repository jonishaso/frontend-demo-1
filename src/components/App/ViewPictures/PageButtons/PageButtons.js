import Button from '@material-ui/core/Button'
import ButtonGroup from '@material-ui/core/ButtonGroup'
import Grid from '@material-ui/core/Grid'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'

const useStyles = makeStyles({
  root: {
    margin: '20px auto'
  }
})

const PageButtons = ({
  disabled,
  currentPage,
  totalPages,
  onNextClick,
  onPreviousClick
}) => {
  const classes = useStyles()

  return (
    <Grid
      className={classes.root}
      container
      alignItems="center"
      justify="center"
    >
      <Grid item>
        <ButtonGroup color="primary" aria-label="outlined primary button group">
          <Button
            disabled={disabled || currentPage == 1}
            onClick={onPreviousClick}
          >
            {'<<'}
          </Button>
          <Button
            disabled={disabled || currentPage == totalPages}
            onClick={onNextClick}
          >
            {'>>'}
          </Button>
        </ButtonGroup>
      </Grid>
    </Grid>
  )
}

export default PageButtons
