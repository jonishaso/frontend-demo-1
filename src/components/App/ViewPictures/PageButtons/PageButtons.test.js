import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PageButtons from './index'

test('test Page Buttons', () => {
  const wrapper = shallow(<PageButtons />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
