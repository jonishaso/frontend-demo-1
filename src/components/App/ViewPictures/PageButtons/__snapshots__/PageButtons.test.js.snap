// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Page Buttons 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="makeStyles-root-1"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(ButtonGroup))
      aria-label="outlined primary button group"
      color="primary"
    >
      <WithStyles(ForwardRef(Button))
        disabled={false}
      >
        &lt;&lt;
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        disabled={true}
      >
        &gt;&gt;
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(ButtonGroup))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
